src/Symfony/Component/Cache/Adapter/RedisAdapter.php

```diff
@@ -19,30 +19,34 @@
 class RedisAdapter extends AbstractAdapter
 {
     private $redis;
+    private $namespace;
 
     public function __construct(\Redis $redisConnection, $namespace = '', $defaultLifetime = 0)
     {
-        $this->redis = $redisConnection;
-
         if (preg_match('#[^-+_.A-Za-z0-9]#', $namespace, $match)) {
             throw new InvalidArgumentException(sprintf('RedisAdapter namespace contains "%s" but only characters in [-+_.A-Za-z0-9] are allowed.', $match[0]));
         }
+        $this->redis = $redisConnection;
+        $this->redis->setOption(\Redis::OPT_PREFIX, $namespace);
```

Comment:
this breaks things if the same Redis connection is reused for several things, as you change its state